poh/src/poh_service.rs

```diff
@@ -543,7 +543,9 @@ impl PohService {
                 }
 
                 // Check if we can break the inner loop to handle a service message.
-                if Self::can_process_service_message(&service_message, &record_receiver) {
+                if next_record.is_none()
```

Comment:
nit: wdyt to include `next_record` in `can_process_service_message()` so the logic of "can break inner loop" is documented in one place?

Reply:
Other users of `can_process_service_message` don't have `next_record` unfortunately. I think the whole flow is a bit hard to follow but I think the goal here is to just fix it and not refactor due to Alpenglow making most of this redundant.